src/core/Meganav.tsx

```diff
@@ -107,147 +23,84 @@ export type MeganavNoticeProps = {
+  const mobileNavItems = useMemo(
+    () =>
+      menuItemsForHeader
+        .filter((item) => !item.isHiddenMobile)
+        .map(({ name, link, content }) => ({ name, link, content })),
+    [],
   );
-};
-
-const SignInPlaceholder = () => <div />;
-
-const panels = {
-  MeganavContentProducts,
-  MeganavContentUseCases,
-  MeganavContentCompany,
-  MeganavContentDevelopers,
-};
-
-const Meganav = ({
-  paths,
-  themeName = "white",
-  notice,
-  loginLink = "/login",
-  urlBase,
-  addSearchApiKey,
-  statusUrl,
-  searchDataId,
-}: MeganavProps) => {
-  const [sessionState, setSessionState] = useState<MeganavSessionState>();
 
   useEffect(() => {
-    // Note if state is never updated, sessionState stays null and never removes the placeholder.
-    // This makes SSR consistent (ie. we always show the placeholder)
-    connectState(selectSessionData, setSessionState);
+    const observeNoticeResize = () => {
+      const noticeElement = document.querySelector('[data-id="ui-notice"]');
+      if (noticeElement) {
+        setNoticeHeight(noticeElement.getBoundingClientRect().height);
+      }
+    };
+    observeNoticeResize();
+    window.addEventListener("resize", observeNoticeResize);
+    return () => window.removeEventListener("resize", observeNoticeResize);
   }, []);
 
-  useEffect(() => {
-    const teardown = MeganavScripts({ themeName, addSearchApiKey });
-    return () => teardown();
-  }, [sessionState]);
-
-  const theme = MeganavData.themes[themeName] as MeganavTheme;
-  const absUrl = (path: string) => _absUrl(path, urlBase);
-
   return (
-    <nav
-      className={`ui-meganav-wrapper ${theme.backgroundColor} ${theme.barShadow}`}
-      data-id="meganav"
-      aria-label="Main"
-    >
-      {notice && <Notice {...notice.props} config={notice.config} />}
-      <div className="ui-meganav ui-grid-px">
-        <div className="mr-24">
-          <Logo dataId="meganav-logo" href={urlBase} logoUrl={paths?.logo} />
-        </div>
-
-        <MeganavItemsDesktop
-          panels={panels}
-          paths={paths}
-          theme={theme}
-          absUrl={absUrl}
-          statusUrl={statusUrl}
-        />
-
-        {/* Because we load the session state through fetch, we display a placeholder until fetch returns */}
-        {sessionState ? (
-          <SignIn
-            sessionState={sessionState}
-            theme={theme}
-            loginLink={loginLink}
-            absUrl={absUrl}
-            searchDataId={searchDataId}
-          />
-        ) : (
-          <SignInPlaceholder />
-        )}
-
-        <MeganavItemsMobile
-          panels={panels}
+    <>
+      <div
+        className="absolute inset-0 w-full z-50"
+        id={theme === "dark" ? "meganav-theme-dark" : "meganav"}
+        data-testid="meganav"
+        style={{ height: HEADER_HEIGHT + noticeHeight }}
+      >
+        {notice && <Notice {...notice.props} config={notice.config} />}
+        <Header
```

Comment:
There's an issue with the z index across the mobile menu and the header where the header bar sits under the underlay that the menu creates. This should have been fixed in [this PR](https://github.com/ably/ably-ui/pull/703), but shout if it isn't.

Reply:
thank you